test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -1339,6 +1342,248 @@ defmodule Trento.SapSystemTest do
   end
 
   describe "deregistration" do
+    test "should deregister a Database and SAP system when the Primary database instance is removed" do
+      sap_system_id = UUID.uuid4()
+      host_id = UUID.uuid4()
+      secondary_database_host_id = UUID.uuid4()
+      application_host_id = UUID.uuid4()
+      deregistered_at = DateTime.utc_now()
+      instance_number_1 = "00"
+      instance_number_2 = "01"
+      application_instance_number = "00"
+
+      assert_events_and_state(
+        [
+          build(
+            :database_registered_event,
+            sap_system_id: sap_system_id
+          ),
+          build(
+            :database_instance_registered_event,
+            sap_system_id: sap_system_id,
+            host_id: host_id,
+            instance_number: instance_number_1,
+            system_replication: "Primary"
+          ),
+          build(
+            :database_instance_registered_event,
+            sap_system_id: sap_system_id,
+            host_id: secondary_database_host_id,
+            instance_number: instance_number_2,
+            system_replication: "Secondary"
+          ),
+          build(
```

Comment:
I would say this test has a "mistake".
It passes, because (emitting the SapSystemDeregistered), because you don't have messageserver/abap combo, not because you remove the database instance